autoinstrumentation/nodejs/src/autoinstrumentation.mts

```diff
@@ -0,0 +1,10 @@
+import "./autoinstrumentation.js";
```

Comment:
@raphael-theriault-swi how does the `--import` flag with with the upstream `opentelemetry/auto-instrumentations-node` ? We are trying to minimize the nodejs code and rely only on that package.


Related ticket: https://github.com/open-telemetry/opentelemetry-operator/issues/3465

Reply:
@trentm could you please take a look here? cc) @JamieDanielson

Reply:
@pavolloffay Are you asking if `node --import @opentelemetry/auto-instrumentations-node/register ...` works? If so, I think yes it should work fine because `node --import something ...` works with `something` being either ESM or CommonJS code. I haven't tried it, though.

However, whether to move the operator to using `@opentelemetry/auto-instrumentations-node` instead of the bootstrap code in "autoinstrumentation.js" doesn't need to be part of *this*  PR, right?

Reply:
> However, whether to move the operator to using @opentelemetry/auto-instrumentations-node instead of the bootstrap code in "autoinstrumentation.js" doesn't need to be part of this PR, right?

The issue is that, if we keep adding new features here it will be harder to migrate them to `auto-instrumentations-node` (or other node packages)

Reply:
Like I mentioned above switching to `auto-instrumentations-node` would be not require any change to the code added by this PR now or in the future.